Editor/Utility/CinemachineEditorAnalytics.cs

```diff
@@ -0,0 +1,252 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEditor;
+using UnityEngine;
+
+namespace Cinemachine.Editor
+{
+    [InitializeOnLoad]
+    static class CinemachineEditorAnalytics
+    {
+        const int k_MaxEventsPerHour = 1000;
+        const int k_MaxNumberOfElements = 1000;
+        const string k_VendorKey = "unity.cinemachine";
+
+        static List<Type> TypesDefinedInCinemachineNamespace;
+
+        // register an event handler when the class is initialized
+        static CinemachineEditorAnalytics()
+        {
+            EditorApplication.playModeStateChanged += SendAnalyticsOnPlayEnter;
+            
+            // Query for all types in cinemachine namespace
+            TypesDefinedInCinemachineNamespace = AppDomain.CurrentDomain.GetAssemblies()
```

Comment:
This isn't what I had in mind.  People can make their own custom types and put them in the "Cinemachine" namespace.  Nothing stops them from doing that.  However, they can't put custom types in the Cinemachine assembly - so that's what you have to check for, not the namespace

Reply:
see https://docs.microsoft.com/en-us/dotnet/api/system.type.assembly?view=net-5.0